src/lib/sum/index.ts

```diff
@@ -0,0 +1,66 @@
+import {PluginInterface} from '../../interfaces';
+import {PluginParams} from '../../types/common';
+import {buildErrorMessage} from '../../util/helpers';
+import {ERRORS} from '../../util/errors';
+const {InputValidationError} = ERRORS;
+
+type sumConfig = {
+  inputParameters: string[];
+  outputParameter: string;
+};
+
+export const Sum = (globalConfig: sumConfig): PluginInterface => {
+  const errorBuilder = buildErrorMessage(Sum.name);
+  const inputParameters = globalConfig.inputParameters;
+  const outputParameter = globalConfig.outputParameter;
+  const metadata = {
+    kind: 'execute',
+  };
+
+  /**
+   * Checks for required fields in input.
+   */
+  const validateSingleInput = (input: PluginParams) => {
+    inputParameters.forEach(metricToSum => {
```

Comment:
it will throw an error if the `inputParameters`, will be better to set an empty array when initializing it.